apps/sms/views/inbox/js/inbox.js

```diff
@@ -585,29 +568,29 @@ var InboxView = {
     // Request and render all threads with drafts
     // or thread-less drafts.
     return Drafts.request(force).then(() => {
-      Drafts.forEach(function(draft, threadId) {
-        var thread;
-        if (threadId) {
+      for (var draft of Drafts.getAll()) {
+        var node = document.getElementById(
+          `thread-${draft.threadId || draft.id}`
+        );
+
+        if (draft.threadId) {
           // Find draft-containing threads that have already been rendered
           // and update them so they mark themselves appropriately
-          var el = document.getElementById('thread-' + threadId);
-          if (el) {
-            thread = Threads.get(threadId);
-            this.updateThread(thread);
+          if (node) {
+            this.updateThread(Threads.get(draft.threadId));
           }
         } else {
           // Safely assume there is a threadless draft
           this.setEmpty(false);
 
-          // If there is currently no list item rendered for this
-          // draft, then proceed.
-          if (!this.draftRegistry[draft.id]) {
-            thread = Thread.create(draft);
+          // Render only in case we don't yet have node for this draft.
+          if (!node) {
```

Comment:
can't we do this without doing a DOM check? Maybe from Threads (because we "set" the draft.id when we load it, maybe we can get it like this?)